pkg/apitype/stacks.go

```diff
@@ -53,6 +56,11 @@ type EncryptValueRequest struct {
 	Plaintext []byte `json:"plaintext"`
 }
 
+type CurrentUpdateStatus struct {
```

Comment:
It is unfortunate that that the name `UpdateStatus` was already taken. But IMHO `CurrentUpdateStatus` doesn't quite smell right.

- "Current" is odd, because it's possible the object is referring to an update that has since completed.
- "Update" is probably OK, but can be a little misleading because of an ambiguity in how we refer to things in the codebase. (Any stack operation is referred to as an "update", but it can take different forms, such as an "import", "preview", "destroy", or... "update".) 

In fact, we should probably add `UpdateKind` to the struct too, since it would be useful.

What do you think of:

```go
	// CurentOperation provides information about any stack operation in-progress as applicable.
 	CurentOperation *OperationStatus `json:"currentUpdate,omitempty"`

...

// OperationStatus describes the state of an operation being performed on a Pulumi stack.
type OperationStatus struct {
   ...   
   Kind UpdateKind `json:"kind"`
    ...
}
```